lib/mix/lib/mix/dep/lock.ex

```diff
@@ -34,6 +34,7 @@ defmodule Mix.Dep.Lock do
   def read() do
     case File.read(lockfile) do
       {:ok, info} ->
+        check_lockfile_conflicts(lockfile, info)
```

Comment:
What about the name `assert_no_merge_conflicts_in_lockfile` as the name for this function?